### PR TITLE
Make restApiVersion on XContentBuilder final

### PR DESCRIPTION
**Pull request by pgomulka**
When passing in restApiVersion during creation of XContentBuilder
it makes it more clear that this field is final.
This prevents accidental change of the version during the xcontent
creation.
The withCompatibleVersion method can also be removed, since the field
only needs to be set in constructor.

relates https://github.com/elastic/elasticsearch/issues/51816

### CODE REVIEW

**Comment by pgomulka** [comment]
@elasticmachine update branch

File: libs/x-content/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java [hunk 1]
```diff
@@ -188,13 +198,15 @@ public XContentBuilder(XContentType xContentType, OutputStream bos, Set<String>
      * @param includes the inclusive filters: only fields and objects that match the inclusive filters will be written to the output.
      * @param excludes the exclusive filters: only fields and objects that don't match the exclusive filters will be written to the output.
      * @param responseContentType  a content-type header value to be send back on a response
+     * @param restApiVersion
```
Comment (jakelandis) [inline libs/x-content/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java:201]: nit: javadoc

File: x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/common/http/HttpRequest.java [hunk 2]
```diff
@@ -511,7 +512,7 @@ public static InputStream filterToXContent(HttpRequest request, XContentType xCo
         try (ByteArrayOutputStream bos = new ByteArrayOutputStream();
              XContentBuilder filteredBuilder = new XContentBuilder(xContentType.xContent(), bos,
                  Collections.emptySet(), Collections.singleton(excludeField),
-                 xContentType.toParsedMediaType())) {
+                 xContentType.toParsedMediaType(), RestApiVersion.current())) {
```
Comment (jakelandis) [inline x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/common/http/HttpRequest.java:515]: I hate the idea of *more* constructors ... but I wonder if it makes sense here to de-emphasize that RestApiVersion is non-consequential. 

The additional constructor, would be the original constructor prior to the PR that would  set RestAPIVersion to current(). That way use cases like this (and enrich) that don't care about REST API compatibility don't need to concern themselves here.

**Review by jakelandis — APPROVED** [review]
Changes look good, I like that you were able to find an non-intrusrive solution here. 

One nitpick about java doc, and I would have (very) minor preference to expose an additional constructor (i.e. preserve the original and default current API version) so that the cases that don't care about REST API compatibility can remain blissfully ignorant of it.

**Pull request merged**